pkg/authenticator/authenticator.go

```diff
@@ -238,14 +240,7 @@ func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
 		content = response
 	}
 
-	// InfoLogger.Printf("Writing token %v to shared volume ...", content)
-	// Create the directory if it doesn't exist
-	tokenDir := filepath.Dir(auth.Config.TokenFilePath)
-	if _, err := os.Stat(tokenDir); os.IsNotExist(err) {
-		os.MkdirAll(tokenDir, 755)
-	}
-
-	err = ioutil.WriteFile(auth.Config.TokenFilePath, content, 0644)
+	err = auth.AccessToken.Write(content)
```

Comment:
why did we switch out `WriteFile` to just `Write`? How do we know the token file path here?

Reply:
why are we writing the access token twice? once here for authn and once for access token file?